frontend/packages/knative-plugin/src/topology/knative-topology-utils.ts

```diff
@@ -826,6 +838,34 @@ export const getSubscriptionTopologyEdgeItems = (
   return edges;
 };
 
+export const getKnSourceKafkaTopologyEdgeItems = (
+  kafkaSource: K8sResourceKind,
+  managedKafkaConnections,
+): EdgeModel[] => {
+  const { data } = managedKafkaConnections;
+  const edges = data.reduce((acc, managedKafkaConnection) => {
+    const mkcServiceAccountSecretName =
+      managedKafkaConnection?.spec?.credentials?.serviceAccountSecretName;
```

Comment:
I think that we should check state of managedkafka CR before creating or we could show edges as pending if we do not want to do it. Both will be trivial to do once we integrate with original PR

Reply:
Ignore. In current implementation we will always make sure that ManagedKafka is valid